src/_pytest/monkeypatch.py

```diff
@@ -336,7 +341,7 @@ def delenv(self, name: str, raising: bool = True) -> None:
         Raises ``KeyError`` if it does not exist, unless ``raising`` is set to
         False.
         """
-        environ: MutableMapping[str, str] = os.environ
+        environ: Mapping[str, str] = os.environ
```

Comment:
Let's keep this unchanged, MutableMapping is still more correct after all...

```suggestion
        environ: MutableMapping[str, str] = os.environ
```